src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKPainter.java

```diff
@@ -965,25 +965,27 @@ public void paintTabbedPaneTabBackground(SynthContext context,
                                            Graphics g,
                                            int x, int y, int w, int h,
                                            int tabIndex) {
-        Region id = context.getRegion();
-        int state = context.getComponentState();
-        int gtkState = ((state & SynthConstants.SELECTED) != 0 ?
-            SynthConstants.ENABLED : SynthConstants.PRESSED);
         JTabbedPane pane = (JTabbedPane)context.getComponent();
-        int placement = pane.getTabPlacement();
+        if (UIManager.getBoolean("TabbedPane.tabsOpaque") || pane.isOpaque()) {
+            Region id = context.getRegion();
+            int state = context.getComponentState();
+            int gtkState = ((state & SynthConstants.SELECTED) != 0 ?
+                    SynthConstants.ENABLED : SynthConstants.PRESSED);
+            int placement = pane.getTabPlacement();
 
-        // Fill the tab rect area
-        g.fillRect(x, y, w, h);
+            // Fill the tab rect area
+            g.fillRect(x, y, w, h);
```

Comment:
I see that the original fix added these lines to GTKPainter. If the fix is being moved individual L&F files is this still needed?

Reply:
Yeah , it is still needed to paint the tab area correctly. Only the condition check part is being moved to individual L&F files in this PR.